File: cpp/film.cpp
#include <iostream>
#include <string>

using namespace std;

// Class Film
class Film {
private:
    string id;
    string judul;
    string genre;
    int harga;

public:
    Film() {}

    Film(string id, string judul, string genre, int harga) {
        this->id = id;
        this->judul = judul;
        this->genre = genre;
        this->harga = harga;
    }

    ~Film() {} // Destruktor

    // Getter
    string getId() const { return id; }
    string getJudul() const { return judul; }
    string getGenre() const { return genre; }
    int getHarga() const { return harga; }

    // Setter
    void setId(string id) { this->id = id; }
    void setJudul(string judul) { this->judul = judul; }
    void setGenre(string genre) { this->genre = genre; }
    void setHarga(int harga) { this->harga = harga; }
};

File: python/Film.py
# Class Film untuk merepresentasikan data film di bioskop
class Film:
    # Constructor dengan atribut private (Encapsulation menggunakan double underscore __)
    def __init__(self, id_film="", judul="", genre="", harga=0):
        self.__id = id_film
        self.__judul = judul
        self.__genre = genre
        self.__harga = harga

    # Getter
    def get_id(self):
        return self.__id

    def get_judul(self):
        return self.__judul

    def get_genre(self):
        return self.__genre

    def get_harga(self):
        return self.__harga

    # Setter
    def set_id(self, id_film):
        self.__id = id_film

    def set_judul(self, judul):
        self.__judul = judul

    def set_genre(self, genre):
        self.__genre = genre

    def set_harga(self, harga):
        self.__harga = harga